test/org/sagebionetworks/bridge/play/controllers/UserManagementControllerTest.java

```diff
@@ -87,6 +116,38 @@ public void before() throws Exception {
         when(request.headers()).thenReturn(map);
         doReturn(null).when(controller).getMetrics();
     }
+
+    @Test
+    public void signInForAdmin() throws Exception {
+        SignIn signIn = new SignIn.Builder().withStudy("originalStudy")
+                .withEmail(TestConstants.EMAIL).withPassword("password").build();
+        TestUtils.mockPlayContextWithJson(signIn);
+
+        when(authService.signIn(eq(study), any(CriteriaContext.class), signInCaptor.capture())).thenReturn(session);
+        
+        Result result = controller.signInForAdmin();
+        TestUtils.assertResult(result, 200);
+        
+        // This isn't in the session that is returned to the user, but verify it has been changed
+        assertEquals("originalStudy", session.getStudyIdentifier().getIdentifier());
+        assertEquals("api", signInCaptor.getValue().getStudyId());
```

Comment:
Do we also want to verify sessionUpdateService was called and cookies are created?

Reply:
Cache provider is a mock dependency of the sessionUpdateService, which was injected, so verifying the cache is called and the session object is updated, verifies all the work of the sessionUpdateService. Added test that cookie is set during this sign in method, in the Local environment.